### PR TITLE
Qa janet azure ocean api

### PR DESCRIPTION
**Pull request by janetlinkiruba**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : Yes
- I have compiled the code successfully my laptop : Yes
- I have written examples to demonstrates the usage of the newly implemented APIs : Yes
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterAdditionalIpConfigurationsAks.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.spotinst.sdkjava.client.rest.IPartialUpdateEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiImportClusterAdditionalIpConfigurationsAks implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterAdditionalIpConfigurationsAks.java:15]: Didn't find this object in the API Documentation.

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterAutoScalerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    @JsonIgnore
+    private Set<String>                                 isSet;
+    private Boolean                                     isEnabled;
+    private ApiImportClusterResourceLimitsSpecification resourceLimits;
+    private ApiImportClusterDownSpecification           down;
+    private ApiImportClusterHeadroomSpecification       headroom;
+    private ApiImportClusterStrategyConfiguration       strategy;
+    private ApiImportClusterHealth                      health;
+    private ApiImportClusterVirtualNodeGroupTemplate    virtualNodeGroupTemplate;
+    private Date                                        createdAt;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterAutoScalerConfiguration.java:27]: createdAt and updatedAt - should this be part of the configuration?

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterLaunchSpecificationAks.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+import com.spotinst.sdkjava.model.bl.ocean.aks.*;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiImportClusterLaunchSpecificationAks implements IPartialUpdateEntity {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterLaunchSpecificationAks.java:17]: Can we update the class-name to reflect that it is a response object.

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterLoginAks.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    public void setUserName(String userName) {
+
+        isSet.add("userName");
+        this.userName = userName;
+    }
+
+    @JsonIgnore
+    public boolean isSshPublicKeySet() {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterLoginAks.java:44]: This function is not required.

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterOsDiskAks.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiImportClusterOsDiskAks implements IPartialUpdateEntity {
+    @JsonIgnore
+    private Set<String>                         isSet;
+    private Integer                             sizeGB;
+    private String                              type;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterOsDiskAks.java:19]: should be defined as enumeration.

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAdditionalIpConfigurationsAks.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
+package com.spotinst.sdkjava.model.bl.ocean.aks;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ImportClusterAdditionalIpConfigurationsAks {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAdditionalIpConfigurationsAks.java:8]: didn't find this class.

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAutomatic.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
+package com.spotinst.sdkjava.model.bl.ocean.aks;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ImportClusterAutomatic {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAutomatic.java:8]: can we rename it to ImportClusterHeadroomAutomatic. Same for API Class.

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterDownSpecification.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ImportClusterDownSpecification {
+
+    @JsonIgnore
+    private Set<String> isSet;
+    private Integer     maxScaleDownPercentage;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterDownSpecification.java:12]: This should be double. The value can be in percentages. Same for API class.

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportOceanClusterAksResponse.java [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+    private Set<String>                                   isSet;
+    private String                                        id;
+    private ClusterConfigurationAks                       aks;
+    private String                                        controllerClusterId;
+    private String                                        name;
+    private ImportClusterAutoScalerConfiguration          autoScaler;
+    private ImportClusterStrategyConfiguration            strategy;
+    private ImportClusterHealth                           health;
+    private ImportClusterVirtualNodeGroupTemplateResponse virtualNodeGroupTemplate;
+    private Date                                          createdAt;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportOceanClusterAksResponse.java:26]: do we need to handle createAt and deletedAt parameters?

**Review by anuragsharma-123 — CHANGES_REQUESTED** [review]
Please review the comments and incorporate the valid ones.

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAdditionalIpConfigurationsAks.java (hunk 6 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAdditionalIpConfigurationsAks.java:8]: This object is in the response, update the class name

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterLaunchSpecificationAks.java (hunk 3 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterLaunchSpecificationAks.java:17]: updated

File: src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterOsDiskAks.java (hunk 5 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/api/ocean/aks/ApiImportClusterOsDiskAks.java:19]: updated

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAutomatic.java (hunk 7 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterAutomatic.java:8]: updated for both

File: src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterDownSpecification.java (hunk 8 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/bl/ocean/aks/ImportClusterDownSpecification.java:12]: updated in both the classes

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**